app/controllers/nonsipp/declaration/SummaryController.scala

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright 2025 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.nonsipp.declaration
+
+import services._
+import viewmodels.implicits._
+import play.api.mvc._
+import utils.nonsipp.summary._
+import controllers.PSRController
+import cats.implicits.{toShow, _}
+import controllers.actions.IdentifyAndRequireData
+import play.api.i18n._
+import models.requests.DataRequest
+import cats.data.EitherT
+import views.html.SummaryView
+import models.SchemeId.Srn
+import pages.nonsipp.CompilationOrSubmissionDatePage
+import play.api.Logger
+import utils.nonsipp.TaskListUtils
+import models.backend.responses.ReportStatus.SubmittedAndSuccessfullyProcessed
+import uk.gov.hmrc.http.HeaderCarrier
+import utils.DateTimeUtils.localDateShow
+import models.{DateRange, NormalMode}
+import viewmodels.DisplayMessage.Message
+import viewmodels.models.{SummaryPageEntry, _}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+import java.time.format.DateTimeFormatter
+import javax.inject.Inject
+
+class SummaryController @Inject() (
+  override val messagesApi: MessagesApi,
+  identifyAndRequireData: IdentifyAndRequireData,
+  val controllerComponents: MessagesControllerComponents,
+  view: SummaryView,
+  val schemeDateService: SchemeDateService,
+  val saveService: SaveService,
+  val psrVersionsService: PsrVersionsService,
+  val psrRetrievalService: PsrRetrievalService
+)(implicit ec: ExecutionContext)
+    extends PSRController
+    with I18nSupport {
+
+  val logger = Logger(getClass)
```

Comment:
Not used

Reply:
used in onPageLoadPreSubmission when the return is not ready for declaration (around line 191)

Reply:
Sorry, my bad.